Merge main to v23.07 branch

dhruvpt opened the pull request.

venkateshpnv commented on pkg/inframanager/p4/cni.go:
```diff
@@ -75,36 +75,31 @@ func ArptToPortTable(ctx context.Context, p4RtC *client.Client, arpTpa string, p
-		err = fmt.Errorf("Invalid IP Address")
-		return err
-	}
+	P4w = GetP4Wrapper(Env)
```
What is the purpose of the wrapper?

venkateshpnv commented on pkg/inframanager/p4/cni.go:
```diff
@@ -75,36 +75,31 @@ func ArptToPortTable(ctx context.Context, p4RtC *client.Client, arpTpa string, p
-			log.Errorf("Invalid mac address %s", macAddr)
-			return err
-		}
+		_ = err
```
the err variable is already used in if condition. I think you don't need this statement.
Do you mean _ = dmac?
In that case, you can do _, err := net.ParseMAC(macAddr).
The checkmarx errors are only visible if the explicit error values returned are not checked.

venkateshpnv commented on pkg/inframanager/p4/cni.go:
```diff
@@ -123,26 +118,49 @@ func Ipv4ToPortTable(ctx context.Context, p4RtC *client.Client, ipAddr string, m
+		err = fmt.Errorf("Invalid MAC Address")
+		return err
+	}
+	_ = macAddr
```
Same here. You can do _, err := net.ParseMAC(mac)

nupurjai replied:
yes, let's avoid it. I think coverity will pickup if error is not handled.

venkateshpnv commented on pkg/inframanager/p4/cni.go:
```diff
@@ -123,26 +118,49 @@ func Ipv4ToPortTable(ctx context.Context, p4RtC *client.Client, ipAddr string, m
-	err := ArptToPortTable(ctx, p4RtC, ipAddr, 0, false)
+func DeleteCniRules(ctx context.Context, p4RtC *client.Client, mac string, ipAddr string) error {
+	var err error
+	Env = "dev"
```
Why do we need to set Env to "dev" here?

nupurjai commented on pkg/inframanager/p4/p4_suite_test.go:
```diff
@@ -0,0 +1,589 @@
+				Expect(ret).Should(BeNil())
+			})
+
+			It("returns err if insert to table fails", func() {
```
I think, I am not sure of the error retun here.

nupurjai replied:
I understood, it's duplicate insert.

nupurjai commented:
Rework in another PR

venkateshpnv committed:
Fixed infraagent failure in insecure mode (#52)

Signed-off-by: Neelakanta Venkatesh Petla <neelakanta.venkatesh.petla@intel.com>
Signed-off-by: Tripathi, Dhruv Prakash <dhruv.tripathi@intel.com>

Commit:
Dev/dhruvpra/k8s merge ipu target (#55)

* Merged files to add support for IPU target

Signed-off-by: Tripathi, Dhruv Prakash <dhruv.tripathi@intel.com>

* Remove IPU P4 artifacts, update Makefile for DPDK target and rename valueToBytes16 as ValueToBytes16 to make it global scope

Signed-off-by: Tripathi, Dhruv Prakash <dhruv.tripathi@intel.com>

* Reset the github actions to same as before for DPDK target.
Created target specific run-test.sh and infraagent-configmap.yaml files and copied the right one at build time.
Updated the inframanager-daemonset.yaml to remove log volume mounts.
Removed the iproute2-rdma from agent Docketfile.
Added mev tag to p4mev_suite_test.go file

Signed-off-by: Neelakanta Venkatesh Petla <neelakanta.venkatesh.petla@intel.com>

* Global renaming and replacement of mev with es2k. Reverted removal of iproute2-rdma from agent docket image

Signed-off-by: Neelakanta Venkatesh Petla <neelakanta.venkatesh.petla@intel.com>

* Revert back to ubuntu-latest for github actions

Signed-off-by: Neelakanta Venkatesh Petla <neelakanta.venkatesh.petla@intel.com>

---------

Signed-off-by: Tripathi, Dhruv Prakash <dhruv.tripathi@intel.com>
Signed-off-by: Neelakanta Venkatesh Petla <neelakanta.venkatesh.petla@intel.com>
Co-authored-by: Neelakanta Venkatesh Petla <neelakanta.venkatesh.petla@intel.com>
Signed-off-by: Tripathi, Dhruv Prakash <dhruv.tripathi@intel.com>

venkateshpnv committed:
Inframanager fixes for code merge. (#56)

* Infrmanager fixes for code merge.

Signed-off-by: Neelakanta Venkatesh Petla <neelakanta.venkatesh.petla@intel.com>

* Set default interface to cdq and mtu size to 1500

Signed-off-by: Neelakanta Venkatesh Petla <neelakanta.venkatesh.petla@intel.com>

* Retain old default arp-mac

Signed-off-by: Neelakanta Venkatesh Petla <neelakanta.venkatesh.petla@intel.com>

---------

Signed-off-by: Neelakanta Venkatesh Petla <neelakanta.venkatesh.petla@intel.com>
Signed-off-by: Tripathi, Dhruv Prakash <dhruv.tripathi@intel.com>

Commit:
Upgrade to Go 1.20.5 (#57)

* Upgrade to Go 1.20.5

Signed-off-by: Tripathi, Dhruv Prakash <dhruv.tripathi@intel.com>

* Run git workflows for DPDK

Signed-off-by: Tripathi, Dhruv Prakash <dhruv.tripathi@intel.com>

---------

Signed-off-by: Tripathi, Dhruv Prakash <dhruv.tripathi@intel.com>

mgaravin committed:
added ipdk_v* branch for github action checks

Signed-off-by: Aravindkumar Mg <aravindkumar.mg@intel.com>

mgaravin reviewed (approved):
Looks good!
